### PR TITLE
Supports ARM Platforms without needing to disable PcdEnforceSecureRngAlgorithms

### PR DESCRIPTION
**Pull request by Flickdm**
## Description

This adds two additional options for "Secure RNG Algorithms" 

```c
 &gEfiRngAlgorithmArmRndr,           // unspecified SP800-90A DRBG via ARM RNDR register
 &gEfiRngAlgorithmRaw                  // Raw data from a NRBG (or a TRNG)
```

gEfiRngAlgorithmRaw takes data from the Hardware or CPU instruction
gEfiRngAlgorithmArmRndr is a newly proposed (and accepted) Guid that gets a "unspecified" SP800-90A algorithm from the CPU

- [X] Impacts functionality?
  - Adds two new acceptable algorithms to NetworkPkg Secure list
- [X] Impacts security?
  - Allows for additional ARM specific algorithm and RAW
- [ ] Breaking change?

- [ ] Includes tests?

- [ ] Includes documentation?

## How This Was Tested

Tested on ARM platform

## Integration Instructions

N/A

### CODE REVIEW

**Review by Javagedes — APPROVED** [review]

**Review by os-d — APPROVED** [review]

File: NetworkPkg/Library/DxeNetLib/DxeNetLib.c
Comment (makubacki) [inline NetworkPkg/Library/DxeNetLib/DxeNetLib.c:1]: @Flickdm, please fill in the PR description.

File: NetworkPkg/Library/DxeNetLib/DxeNetLib.c [hunk 1]
```diff
@@ -139,9 +139,11 @@ GLOBAL_REMOVE_IF_UNREFERENCED VLAN_DEVICE_PATH  mNetVlanDevicePathTemplate = {
 // then you should place that algorithm first in the array.
 //
 GLOBAL_REMOVE_IF_UNREFERENCED EFI_GUID  *mSecureHashAlgorithms[] = {
-  &gEfiRngAlgorithmSp80090Ctr256Guid,  // SP800-90A DRBG CTR using AES-256
-  &gEfiRngAlgorithmSp80090Hmac256Guid, // SP800-90A DRBG HMAC using SHA-256
-  &gEfiRngAlgorithmSp80090Hash256Guid  // SP800-90A DRBG Hash using SHA-256
+  &gEfiRngAlgorithmSp80090Ctr256Guid,   // SP800-90A DRBG CTR using AES-256
+  &gEfiRngAlgorithmSp80090Hmac256Guid,  // SP800-90A DRBG HMAC using SHA-256
+  &gEfiRngAlgorithmSp80090Hash256Guid,  // SP800-90A DRBG Hash using SHA-256
+  &gEfiRngAlgorithmArmRndr,             // unspecified SP800-90A DRBG via ARM RNDR register
```
Comment (makubacki) [inline NetworkPkg/Library/DxeNetLib/DxeNetLib.c]: Was there a particular reason to break formatting? For example, to state:

```suggestion
  &gEfiRngAlgorithmArmRndr,             // SP800-90A DRBG unspecified algorithm via the Arm RNDR register
```

File: NetworkPkg/Library/DxeNetLib/DxeNetLib.c [hunk 2]
```diff
@@ -139,9 +139,11 @@ GLOBAL_REMOVE_IF_UNREFERENCED VLAN_DEVICE_PATH  mNetVlanDevicePathTemplate = {
...
 //
 GLOBAL_REMOVE_IF_UNREFERENCED EFI_GUID  *mSecureHashAlgorithms[] = {
-  &gEfiRngAlgorithmSp80090Ctr256Guid,  // SP800-90A DRBG CTR using AES-256
-  &gEfiRngAlgorithmSp80090Hmac256Guid, // SP800-90A DRBG HMAC using SHA-256
-  &gEfiRngAlgorithmSp80090Hash256Guid  // SP800-90A DRBG Hash using SHA-256
+  &gEfiRngAlgorithmSp80090Ctr256Guid,   // SP800-90A DRBG CTR using AES-256
+  &gEfiRngAlgorithmSp80090Hmac256Guid,  // SP800-90A DRBG HMAC using SHA-256
+  &gEfiRngAlgorithmSp80090Hash256Guid,  // SP800-90A DRBG Hash using SHA-256
+  &gEfiRngAlgorithmArmRndr,             // unspecified SP800-90A DRBG via ARM RNDR register
+  &gEfiRngAlgorithmRaw                  // Raw data from NRBG (or TRNG)
```
Comment (makubacki) [inline NetworkPkg/Library/DxeNetLib/DxeNetLib.c]:
```suggestion
  &gEfiRngAlgorithmRaw                  // Raw data from a NRBG (or a TRNG)
```

**Review by makubacki — APPROVED** [review]

**Review by apop5 — APPROVED** [review]

File: NetworkPkg/Library/DxeNetLib/DxeNetLib.c (hunk 1 above)
Comment (Flickdm) [reply NetworkPkg/Library/DxeNetLib/DxeNetLib.c]: No this was me taking ards change, but I can correct this here and upstream

**Pull request merged**